### PR TITLE
[DO-NOT-MERGE-WIP] VACMS-5673: Adding ga scripts to site to test for push connectivity from va network.

### PR DESCRIPTION
**Pull request by ethanteague**
## Description
See https://github.com/department-of-veterans-affairs/va.gov-team/issues/5673

## Testing done
Will be performed in conjunction with Nedie and Jon (analytics team)

## Screenshots
N/A

## QA steps
N/A (for now)

### CODE REVIEW

File: docroot/modules/custom/va_gov_backend/va_gov_backend.module [hunk 1]
```diff
@@ -232,15 +235,115 @@ function va_gov_backend_inline_entity_form_entity_form_alter(&$entity_form, $for
...
+
+/**
+ * Builds section associations for gtm push array.
+ *
+ * @return array
+ *   Keyed array of the section item names.
+ */
+function _va_gov_backend_gtm_settings() {
+  $sections = [];
+  $page_title = '';
```
Comment (oksana-c) [inline docroot/modules/custom/va_gov_backend/va_gov_backend.module]: unused variable

**Comment by oksana-c** [comment]
@ethanteague, were you able to verify that correct data flows into GTM dataLayer using test account?

File: docroot/modules/custom/va_gov_backend/js/gtm_tag_trackers.js [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+  Drupal.behaviors.vaGovTagTracker = {
+    attach: function (context, settings) {
+      // Build our gtm settings push object.
+      const dataCollection = {
+        'pagePath': settings.gtm_data.pagePath ? settings.gtm_data.pagePath : null,
+        'pageTitle': settings.gtm_data.pageTitle ? settings.gtm_data.pageTitle : null,
+        'firstSectionLevel': settings.gtm_data.firstSectionLevel ? settings.gtm_data.firstSectionLevel : null,
+        'secondSectionLevel': settings.gtm_data.secondSectionLevel ? settings.gtm_data.secondSectionLevel : null,
+        'thirdSectionLevel': settings.gtm_data.thirdSectionLevel ? settings.gtm_data.thirdSectionLevel : null,
+        'fourthSectionLevel': settings.gtm_data.fourthSectionLevel ? settings.gtm_data.foourthSectionLevel : null,
```
[the remark this answers is not in the document]
Comment (ethanteague) [reply docroot/modules/custom/va_gov_backend/js/gtm_tag_trackers.js]: That's a keen eye, @oksana-c - this is something I _never_ would have found in regular qa, as fourth level items are usually not triggered. TY!

**Comment by oksana-c** [comment]
P.S. Behat tests are failing, see screenshot. I think it may be due to the outdated branch.
Screen Shot 2020-03-20 at 1 18 24 PM

File: docroot/modules/custom/va_gov_backend/va_gov_backend.module [hunk 3]
```diff
@@ -5,12 +5,15 @@
...
 
+use Symfony\Cmf\Component\Routing\RouteObjectInterface;
 use Drupal\Core\Asset\AttachedAssetsInterface;
 use Drupal\node\NodeInterface;
 use Drupal\Core\Routing\RouteMatchInterface;
 use Drupal\Core\Field\WidgetBase;
 use Drupal\Core\Form\FormStateInterface;
+use Drupal\Core\Render\Markup;
 use Drupal\Core\Url;
+use Drupal\taxonomy\Entity\Term;
```
[the remark this answers is not in the document]
Comment (ethanteague) [reply docroot/modules/custom/va_gov_backend/va_gov_backend.module]: Nice - will do, @oksana-c

File: docroot/modules/custom/va_gov_backend/va_gov_backend.module (hunk 1 above)
Comment (ethanteague) [reply docroot/modules/custom/va_gov_backend/va_gov_backend.module]: > @ethanteague, were you able to verify that correct data flows into GTM dataLayer using test account?
Yep! I've been testing with Jon & Nedie on the analytics team, @oksana-c

**Comment by ethanteague** [comment]
> P.S. Behat tests are failing, see screenshot. I think it may be due to the outdated branch.
> Screen Shot 2020-03-20 at 1 18 24 PM

Great of you to test that, @oksana-c - I'll take a look :+1:

File: docroot/modules/custom/va_gov_backend/js/gtm_tag_trackers.js [hunk 4]
```diff
@@ -0,0 +1,159 @@
...
+      const gtmPageLoadPush = () => {
+        dataCollection['event'] = 'pageLoad';
+        dataLayer.push(dataCollection);
+      }
+
+      // Three behaviors loaded on edit pages.
+      // This causes function to fire three times on page load.
+      // I don't think complex limiting of behaviors can be done without jQuery.
+      // Google searches all point to jQuery to handle this issue.
+      $(window, context).once('vaGovTagTracker').on('load', function () {
```
Comment (indytechcook) [inline docroot/modules/custom/va_gov_backend/js/gtm_tag_trackers.js:152]: @ethanteague I'm not sure I understand this comment.  Is there a requirement to not use `jQuery`?  

I noticed several places jquery was not selected to be used to work with the DOM.  Was there a reason for this?

jQuery gives will give cross-browser consistency when we are not using any other framework (React/Vue/Angular).

Comment (ethanteague) [reply docroot/modules/custom/va_gov_backend/js/gtm_tag_trackers.js:152]: There's no requirement to not use jQuery. I'm just trying to get out of the habit of using it now that ES6 has just about everything we need to easily manipulate the dom, sans JQ, and JQ is kind of seen as more of an old school way of doing things. I don't think the cross browser support/consistency benefit provided by jQuery is as much of a concern these days, @indytechcook

Comment (indytechcook) [reply docroot/modules/custom/va_gov_backend/js/gtm_tag_trackers.js:152]: Thanks @ethanteague .

>  I don't think the cross browser support/consistency benefit provided by jQuery is as much of a concern these days

Would you mind elaborating on this statement?  It's the opposite of my experience when working with multiple different types of browsers.

Comment (ethanteague) [reply docroot/modules/custom/va_gov_backend/js/gtm_tag_trackers.js:152]: > Thanks @ethanteague .
> 
> > I don't think the cross browser support/consistency benefit provided by jQuery is as much of a concern these days
> 
> Would you mind elaborating on this statement? It's the opposite of my experience when working with multiple different types of browsers.

I just mean that most ES6 methods are widely supported across modern browsers these days, and we aren't concerned with legacy versions of IE on this project, @indytechcook

**Pull request closed without merging**